### ISSUE TITLE
keine Aktualisierungen mehr

### ISSUE THREAD

**Issue by galegro** [PfisterDaniel/iobroker.apple-find-me#37]
Seit heute (ca. 18:00 Uhr, 5. Okt. 2022) werden keine Werte mehr aktualisiert.

Der Fehler tritt nicht nur in unserer Installation auf. Auch in einem weiteren System werden keine Werte mehr aktualisiert. Die letzten Zeitstempel liegen in beiden Systemen nur wenige Minuten auseinander.

In beiden System sind die Adapter offenbar abgestürzt. Ein Neustart (auch des gesamten Systems) änderte nichts an dem Fehler.

**Comment by PfisterDaniel** [comment]
Hallo, ja beim mir ist auch das Problem. Vermutlich hat Apple mal wieder irgendwelche Probleme ich hoffe morgen funktioniert's wieder. Oder Apple hat an sich was an der Schnittstelle geändert.

**Comment by galegro** [comment]
> .. Oder Apple hat an sich was an der Schnittstelle geändert.

Dass in diesem Fall keine Daten mehr abgerufen werden können, leuchtet mir ein. Darüber hinaus stürzen die Adapter bei den von mir genannten Systemen mittlerweile regelmäßig ab.

**Comment by Sandomor** [comment]
Bei mir das gleiche - im Log steht, dass die Credentials ned stimmen - sind überprüft - aber der Adapter kommt nicht mehr auf die Schnittstelle...

und nach einigen Fehlern im Log, beendet sich der Adapter...

**Comment by Zerpac54** [comment]
> Hallo, ja beim mir ist auch das Problem. Vermutlich hat Apple mal wieder irgendwelche Probleme ich hoffe morgen funktioniert's wieder. Oder Apple hat an sich was an der Schnittstelle geändert.

Hallo ich habe das gleiche Problem, habe aber irgendwo gelesen, dass der Adapter bei zweifaktor Authentifizierung nicht funktioniert, ist das korrekT?

**Comment by Zerpac54** [comment]
Hallo ich habe das gleiche Problem, habe aber irgendwo gelesen, dass der Adapter bei Zwei Faktor Authentifizierung nicht funktioniert, ist das korrekt?

**Comment by galegro** [comment]
> .., dass der Adapter bei Zwei Faktor Authentifizierung nicht funktioniert, ..

Die Zwei-Faktor-Authentifizierung nutze ich. Über Wochen arbeitete der Adapter unter diesen Voraussetzungen einwandfrei.

**Comment by Zerpac54** [comment]
Funktioniert er momentan auch noch? Bei mir kommt immer die gleiche Fehlermeldung:

	Error on HTTP-Request. Please check your credentials. StatusCode: 401 Retry in 10 minutes. (2/3)

**Comment by mstiem** [comment]
> Funktioniert er momentan auch noch? Bei mir kommt immer die gleiche Fehlermeldung:
> 
> ```
> Error on HTTP-Request. Please check your credentials. StatusCode: 401 Retry in 10 minutes. (2/3)
> ```

I have the same issue. I already tried to use not my apple ID instead use the apple app password with your id which you can create for each independent app -then 2FA is not required. This also does not work right now.

**Comment by PfisterDaniel** [comment]
Hello, the problem is not 2FA. My requests to https://fmipmobile.icloud.com are not working. I think Apple service is down on this domain. I hope it's temporary and not forever... Maybe Apple changed authentication. I don't know at the moment. We'll have to wait and see for now.

**Comment by Sandomor** [comment]
Hi Daniel,
i think they have changed the Authentication - the InitRequest gives the respond 401 ...

Hoping that is a temporare issue ....

**Comment by PfisterDaniel** [comment]
Yes, we had the same problem a few weeks ago. 2 day later it worked perfectly again.

**Comment by Sandomor** [comment]
so lets hope :)

**Comment by radebold** [comment]
Hi Daniel

Die Server scheinen online zu sein, wenn da nur die webapi oder?

Viele Grüsse
Thomas

Daniel Lippert ***@***.***> schrieb am Do. 6. Okt. 2022 um
20:44:

**Comment by radebold** [comment]
https://www.ssllabs.com/ssltest//analyze.html?d=fmipmobile.icloud.com


Daniel Lippert ***@***.***> schrieb am Do. 6. Okt. 2022 um
20:44:

**Comment by PfisterDaniel** [comment]
@Thomas 
Ja wenn nur die API. Über Apple's FindMe App funktioniert es ja. Die nutzt die selbe Domain nur verwendet Sie einen anderen Authentifierzungs-Mechanismus. Hier lauten die Domains lt. Sniffer p23.fmipmobile.icloud.com usw.

**Comment by mstiem** [comment]
The issue @radebold  is tryling to link to is that all the certificates are not valid at this point in time. Do you expect an https connection. I hope ;) as long as it is not fixed and trused to all .. there will be an issue on your end
![image](https://user-images.githubusercontent.com/2210293/194400161-003318bc-df0e-4be1-bf0a-2fb36758755e.png)

**Comment by Zerpac54** [comment]
Sorry das verstehe ich nicht, was heisst das jetzt?

**Comment by prorun26** [comment]
Das bedeutet dass es aktuell nicht mehr funktioniert und wir leider warten müssen. Ich hoffe jedoch nicht all zu lange.

**Comment by mstiem** [comment]
Hello, 
I'm not so sure anymore. After having a look into the [code](https://github.com/PfisterDaniel/ioBroker.apple-find-me/blob/master/main.js#L116)  and then having a next closer look on the "Handshake Simulation" on the lower part of this [page](https://www.ssllabs.com/ssltest/analyze.html?d=fmipmobile.icloud.com&s=2620%3a149%3aa43%3a10d%3a0%3a0%3a0%3ac&latest) I personally think this should work. 
If it would be an 400 Bad Request error code -> indicating ssl issues but the 401 may seems that really the authentication is the issue.
 If I have time later I will try to use postman on my end and do some trials...

**Comment by PfisterDaniel** [comment]
In Postman is the same 401...
![Bildschirmfoto 2022-10-07 um 14 07 40](https://user-images.githubusercontent.com/9358627/194550139-0b583db7-c711-4c77-8d4c-6dc3015deedd.png)
I checked some other APIs (Python, Node.js), none API is currently working.

**Comment by sleepless135** [comment]
hey,
iam new to the apple and IO Broker World.

Error on HTTP-Request. Please check your credentials. StatusCode: 401 Retry in 5 minutes. (2/3)

Iam an Android User, but want to track my Bike with airtags.

So i find this option, but at the moment i just get errors form the plugin.
Is that the same problem that you have on this bug report ? or i did something wrong in the config

Best greats 
Tim

**Comment by galegro** [comment]
Nun sind Wochen vergangen. Der Adapter funktioniert leider immer noch nicht. Wer diesen Adapter kennt, dürfte ihn sehr vermissen.

Müssen wir weiter auf Apple hoffen oder besteht Hoffnung, dass der Entwickler sich des Ganzen erbarmt?

**Comment by davidcreager** [comment]
Indeed, this library is sorely missed.   Do we know what the actual problem is?  What has apple changed that has stopped this working?

**Comment by Lycia-Renfort** [comment]
Any way we can help to fix this?

**Comment by Knallochse** [comment]
[Openhab Lösung](https://github.com/openhab/openhab-addons/pull/13691)

**Comment by galegro** [comment]
Endlich scheint es weiter zu gehen. Leider kann ich die gewünschten Tests nicht ausführen, da ich schlichtweg zu wenig von dem Ganzen verstehe.

**Comment by galegro** [comment]
Ist das Projekt mittlerweile eingemottet oder tut sich noch etwas?

**Comment by davidcreager** [comment]
Sadly it seems to be mothballed.   It's a shame because I think this is definitely fixable (based on other working repos). I don't have the time to research bbi guess same for everyone

**Comment by Apollon77** [comment]
Yes and somewhen someone also will find time to look into it when the developer itself has no chance ... stay patient

**Comment by galegro** [comment]
Nach Monaten Geduld haben müssen, ist nur noch Ungeduld übrig. Ich habe zwar Zeit, aber nicht das nötige Können.

**Comment by Apollon77** [comment]
Dann musst Du die Ungeduld aushalten ... sorry das ich da keine bessere Nachrichten für dich habe

**Comment by galegro** [comment]
Wenn es doch noch weitergeht, will ich gerne die Geduld aufbringen.

**Comment by Apollon77** [comment]
Sagen wir so: Der Adapter steht auf meiner Liste anzuschauen  und ggf zu übernehmen in dei community ... aber Matter ist gerade wichtiger

**Comment by prorun26** [comment]
Hi Apollon77, wie siehts mittlerweile aus mit der Liste der anzuschauenden und ggf zu übernehmenden Adapter die vor einigen Monaten so viel Hoffnung bereitet hat. Ich spreche glaube ich für Viele, die sich nach wie vor sehr wünschen würden, dass dieser Adapter nachgerutscht ist und vielleicht sogar bald Zuneigung findet. Ich biete mich auf alle Fälle als Tester an.

**Comment by Apollon77** [comment]
"Matter" als Thema steht vorher auf der Liste und das wird (zumindestens mich) noch eine Weile beschäftigen) ...

**Comment by Mittelblut9** [comment]
Hallo zusammen.
ich habe mich vor ca. 1 Monat mit dem Problem beschäftigt und soweit auch eine gute Lösung gefunden, mit der ich den Adapter zum Laufen bekomme. Leider gibt es einen Bug in der Abhängigkeit, die ich benutze, aber das sollte behebbar sein.
Im letzten Monat kam bei mir viel anderes Zeug dazwischen, weshalb ich die Zeit nicht nutzen konnte. Nach meinem jetzigen Urlaub werde ich mich wieder dran setzen.
Ihr findet das Repo auf meinem Profil 🤝

**Comment by radebold** [comment]
> Hallo zusammen. ich habe mich vor ca. 1 Monat mit dem Problem beschäftigt und soweit auch eine gute Lösung gefunden, mit der ich den Adapter zum Laufen bekomme. Leider gibt es einen Bug in der Abhängigkeit, die ich benutze, aber das sollte behebbar sein. Im letzten Monat kam bei mir viel anderes Zeug dazwischen, weshalb ich die Zeit nicht nutzen konnte. Nach meinem jetzigen Urlaub werde ich mich wieder dran setzen. Ihr findet das Repo auf meinem Profil 🤝

@Mittelblut9 was muss man denn tun, um deine "lauffähige Version" zu nutzen?

**Comment by Mittelblut9** [comment]
@radebold noch nicht. Ich bin gerade noch dran. Nach meinem Urlaub werde ich mich wieder dransetzen

**Comment by radebold** [comment]
> @radebold noch nicht. Ich bin gerade noch dran. Nach meinem Urlaub werde ich mich wieder dransetzen

@Mittelblut9 
sehr cool, ich hoffe mal auf eine bald lauffähige Version. Wenn du jemanden zum testen benötigst, sag bescheid.

**Comment by prorun26** [comment]
> > @radebold noch nicht. Ich bin gerade noch dran. Nach meinem Urlaub werde ich mich wieder dransetzen
> 
> @Mittelblut9 sehr cool, ich hoffe mal auf eine bald lauffähige Version. Wenn du jemanden zum testen benötigst, sag bescheid.

Da schließe ich mich gerne an!

**Comment by Mittelblut9** [comment]
Gute Neuigkeiten!
Ich bekomme alle Daten, die derzeit genutzt werden. 🎉 
(Beispiel Daten!)
![grafik](https://github.com/PfisterDaniel/ioBroker.apple-find-me/assets/67608550/95521900-d583-43ae-a8de-1a8c68de8372)

Bevor ich das freigebe zur Nutzung möchte ich jedoch den Code noch von Grund auf refaktorieren um den Adapter für die Zukunft maintainable zu halten.

Die, die den Adapter, nach der Refaktorierung, testen wollen können mich gerne auf Discord adden. 
Discord Name: **mittelblut9**

👍

**Comment by Ilovegym66** [comment]
Super, vielen DANK! Ich hab dein Repo schon laenger geliked und hab den Fortschritt verfolgt, super gemacht! Das hat alles Hand und Fuss, was du da machst.

**Comment by Mittelblut9** [comment]
1 Monat später...
Eigentlich ist der Adapter jetzt auf einer guten Codebasis und so gut wie einsatzbereit. Aber wie das so ist, eine Sache blockiert nun die komplette Entwicklung.
Grob gesagt: Das Package, welches auch schon seit 3 Jahren kein Update mehr bekommen hat, bzw. Apple erzeugt nur ein Cookie, welches nur 10-15 Minuten gültig ist, und ich im Adapter theoretisch den User alle 15 Minuten neu einloggen lasse (automatisch).
Das löst aber alle 15 Minuten eine Benachrichtigungsmail von Apple aus. D.h. an einem Tag würde ein User 96 Mails bekommen (wenn ich richtig gerechnet habe).

Der Versuch das Cookie zu manipulieren oder den Request-Body zu ändern hat beides nicht funktioniert.

Also habe ich die Entwicklung vorerst eingestellt. Vielleicht wird das Package aktualisiert oder ein anderes entwickelt, das auf dem neuesten Stand ist.

Schade. Es wäre sehr cool gewesen, wenn das alles funktioniert hätte.

**Comment by prorun26** [comment]
Oh je, das sind aber sehr ernüchternde Neuigkeiten. Kann der Adapter denn in dieser Form, trotz 96 Mails pro Tag sorgenfrei verwendet werden?

**Comment by Ilovegym66** [comment]
Ich hab ihn fuer 10h laufen, aber einen ganzen Tag... es kann sein, dass dein Account aus Sicherheitsgruenden von Apple blockiert wird.. zur Zeit ist das leider keine dauerhafte Loesung, aber noch ist nicht aller Tage Ende.. :-)

**Comment by ticaki** [comment]
@Mittelblut9 
> Also habe ich die Entwicklung vorerst eingestellt. Vielleicht wird das Package aktualisiert oder ein anderes entwickelt, das auf dem neuesten Stand ist.

https://github.com/foxt/icloud.js

willst du weiter machen?

**Comment by Mittelblut9** [comment]
> @Mittelblut9
> 
> > Also habe ich die Entwicklung vorerst eingestellt. Vielleicht wird das Package aktualisiert oder ein anderes entwickelt, das auf dem neuesten Stand ist.
> 
> https://github.com/foxt/icloud.js
> 
> willst du weiter machen?

Oh nice. Probier ich mal aus 👍 Thx!

**Comment by ticaki** [comment]
Ich hab den Eindruck gewonnen, das alle Denken ich hätte da schon Arbeit rein gesteckt. Das was ich gemacht habe besteht nur aus der Authentifizierung und einer Test Adminoberfläche.

**Comment by Ilovegym66** [comment]
Ich denke, dass man hier eine Loesung hat, um den Adapter wieder zum leben zu erwecken.. :-) 
Und moechte mich als User schonmal bei allen Bedanken, die daran arbeiten!
:-)
Login und Auth ist doch schon super, genau daran hing es.. es kam alle 15 min ne Email von Apple.. wenn das behoben ist, duerfte der Rest "nur noch" Daten holen sein...??

**Comment by ticaki** [comment]
Naja ich hab noch mal ne halbe stunde geguckt, aber nach ca 3 Minuten muß man sich neu anmelden mit email. Ich kram da jetzt nicht weiter rum, weil ich die python lib aus der abgeguckt wird wegen mangelnden Sprachkenntnissen nicht verwenden kann.

**Comment by Mittelblut9** [comment]
![grafik](https://github.com/PfisterDaniel/ioBroker.apple-find-me/assets/67608550/2f242d6a-f121-4dad-8ef5-8eff4f9ff285)
Hmm.. Mein Test code läuft jede Minute und falls ein Fehler auftritt wird der User neu angemeldet... Aber dann passiert wieder genau das was das andere Module auch hatte... Massig E-Mails..

**Comment by Mittelblut9** [comment]
\- ... und der eine Fehler, der jedesmal auftritt ist, dass Apple mir einen 450er Status zurück schickt, der so viel heißt wie: "Melde dich neu an"

**Comment by ticaki** [comment]
Jo, zu dem Fehler hab ich das gefunden https://github.com/foxt/icloud.js/issues/11
Ich finde aber keine issue zu den vielen emails in pyicloud, aber da ist das letzte commit auch ewig her.

das hier ist die integration in hass https://github.com/gcobb321/icloud3 damit bin ich aber überfordert.

**Comment by Mittelblut9** [comment]
> Jo, zu dem Fehler hab ich das gefunden [foxt/icloud.js#11](https://github.com/foxt/icloud.js/issues/11) Ich finde aber keine issue zu den vielen emails in pyicloud, aber da ist das letzte commit auch ewig her.
> 
> das hier ist die integration in hass https://github.com/gcobb321/icloud3 damit bin ich aber überfordert.

Das was in dem Issue steht, hab ich ausprobiert. Funktioniert trz nicht :shrug: 

Das von Homatic ist auch overkill... zig tausend Zeilen code durch schauen ist auch eine Torture

**Comment by ticaki** [comment]
Vielleicht schreibt ja nochmal jemand ne lib - die dann 3 monate geht bis apple wieder was ändert. Ich halte auf jeden Fall die Augen offen.

**Comment by ticaki** [comment]
Der Vollständigkeit halber: Bei Openhab sieht das so aus... https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.icloud

**Comment by mcm1957** [comment]
Ein paar Anmerkungen / Workarounds gibts ev. da
https://forum.iobroker.net/topic/73625/deprecated-abk%C3%BCndigung-adapter-iobroker-apple-find-me/2
